src/server/editorServices.ts

```diff
@@ -1350,6 +1350,22 @@ namespace ts.server {
             }
         }
 
+        /*@internal*/
+        onExtendedConfigChangedForConfiguredProject(project: ConfiguredProject, extendedConfigFile: string) {
+            const configFileExistenceInfo = this.getConfigFileExistenceInfo(project);
+
+            this.logConfigFileWatchUpdate(asNormalizedPath(extendedConfigFile), project.canonicalConfigFilePath, configFileExistenceInfo, ConfigFileWatcherStatus.ReloadingFiles);
+
+            // Skip refresh if project is not yet loaded
+            if (project.isInitialLoadPending()) return;
+            project.pendingReload = ConfigFileProgramReloadLevel.Full;
+            project.pendingReloadReason = "Change in extended config file detected";
+            this.delayUpdateProjectGraph(project);
+            // As we scheduled the update on configured project graph,
+            // we would need to schedule the project reload for only the root of inferred projects
+            this.delayReloadConfiguredProjectForFiles(configFileExistenceInfo, /*ignoreIfNotInferredProjectRoot*/ true);
+        }

```

Comment:
I created this function because the regular config file watcher triggers `onConfigChangedForConfiguredProject` (directly above) and it didn't make sense to call that for the extended configs. This new function still calls `logConfigFileWatchUpdate` (although with the extended config path) and `delayReloadConfiguredProjectForFiles`. Let me know if there's a better strategy for listening to extended config changed events, as I'm not sure if my changes here are correct.